h5py/_hl/dataset.py

```diff
@@ -90,12 +90,17 @@ def make_new_dset(parent, shape=None, dtype=None, data=None,
             data = Empty(dtype)
         shape = data.shape
     else:
-        shape = tuple(shape)
+        shape = tuple((shape,)) if isinstance(shape, int) else tuple(shape)
```

Comment:
I think the first `tuple()` here is unnecessary: `(shape,)` makes a tuple already.